### ISSUE TITLE
Maps API not working on the web for directions drop-down

### ISSUE THREAD

**Issue by werebus** [umts/pvta-multiplatform#487]
![image](https://user-images.githubusercontent.com/240275/48223325-d1268f00-e364-11e8-871a-30d2ef2e0ec4.png)

In the console:

```
This API project is not authorized to use this API. Please ensure this API is activated in the Google Developers Console: https://console.developers.google.com/apis/api/places_backend?project=_ For more information on authentication and Google Maps JavaScript API services please see: https://developers.google.com/maps/documentation/javascript/get-api-key js:76:50
	_.Gc https://maps.googleapis.com/maps/api/js:76:50
	R2/< https://maps.googleapis.com/maps-api-v3/api/js/34/16/places_impl.js:7:62
	O2/< https://maps.googleapis.com/maps-api-v3/api/js/34/16/places_impl.js:5:431
	e https://maps.googleapis.com/maps-api-v3/api/js/34/16/places_impl.js:4:143
	c https://maps.googleapis.com/maps-api-v3/api/js/34/16/common.js:47:82
	<anonymous> https://maps.googleapis.com/maps/api/place/js/AutocompletionService.GetPredictions:1:22
```

I don't _think_ that this is related to #485, but there's a chance.

**Comment by mboneil10** [comment]
Oh, interesting

**Comment by mboneil10** [comment]
@werebus, is this also on iOS like #485?

**Comment by werebus** [comment]
Nope. It's on the web platform (https://m.pvta.com/#/plan-trip).

**Comment by mboneil10** [comment]
Oh, sorry. Reread the title of the issue right now. Thanks

**Comment by mboneil10** [comment]
I've looked into this and fixed the dropdown in development. I was able to do so by adding these APIs: 
- Geocoding
> The Geocoding API is a service that provides geocoding and reverse geocoding of addresses.
> 
> Geocoding is the process of converting addresses (like a street address) into geographic coordinates (like latitude and longitude), which you can use to place markers on a map, or position the map. Reverse geocoding is the process of converting geographic coordinates into a human-readable address.
> 
> You can also use the Geocoding API to find the address for a given place ID.

- Places
> The Places API is a service that returns information about places using HTTP requests. Places are defined within this API as establishments, geographic locations, or prominent points of interest.

I wonder if this _is_ related to #485, since the dropdown on android is also not functioning properly.

**Comment by mboneil10** [comment]
The thing that's stopping me from making this change to the other API keys is that I'm not sure if we need the Geocoding API to be added. Afterall, it is part of the Google Maps JavaScript API (currently the only API we're using on PVTrAck). 

Update: all good! Deleting the Geocoding API made no difference

**Comment by mboneil10** [comment]
The problem was occurring in the autocomplete of the [plan trip/the schedule page](https://github.com/umts/pvta-multiplatform/blob/08f7d1b5628bbb621cc40f44ec1a064a879c8926/src/pages/plan-trip/plan-trip.component.ts#L218
). So autocomplete relies on Places. I enabled that API.

To double check, I looked at the [API picker](https://developers.google.com/maps/documentation/api-picker) on Google Maps Platform, which indicates that Places API includes the following request:
> Place Autocomplete automatically fills in the name and/or address of a place as users type.

**Comment by mboneil10** [comment]
TLDR -- Enabling the Places API on Google Cloud fixes the problem in development. Should I do the same for production?

Please let me know if this sounds like a good idea @sherson, @anbranin, @werebus, and I'll make the proper changes. 👍

**Comment by werebus** [comment]
I think our app [does use the Geocoding API](https://github.com/umts/pvta-multiplatform/blob/08f7d1b5628bbb621cc40f44ec1a064a879c8926/src/pages/plan-trip/plan-trip.component.ts#L76-L100) to encode the user's current location into a place name.

**Comment by werebus** [comment]
Yes, if our app uses a particular API, that API needs to be enabled. And, at least for now, all of the versions of our app — iOS, Android, Web — use the same APIs.

**Comment by mboneil10** [comment]
Got it! Enabling them for production now.

**Comment by mboneil10** [comment]
Errr the autocomplete works on web, but I keep getting this message...
<img width="290" alt="screen shot 2018-11-19 at 9 25 25 am" src="https://user-images.githubusercontent.com/19267771/48713058-633c5c00-ebdd-11e8-9765-25ef852b3cfe.png">

**Comment by mboneil10** [comment]
Enabled Directions API, all good.

**Comment by mboneil10** [comment]
After adding the APIs, looks good to me. Will reopen if there still seems to me an issue.
